src/here-xyz.ts

```diff
@@ -2081,31 +2108,249 @@ function getVirtualSpaceProfiles(title: string, description: string, spaceids: A
+    return tagName && tagName.trim().length > 0 && tagName.indexOf(",") == -1
+}
+
+function isValidRuleExpression(ruleExpression:string) {
+    return ruleExpression && ruleExpression.trim().length > 4
+}
+
+program
+    .command("tagrules <id>")
+    .description("add, remove, view the conditional rules to tag your features automatically, at present all tag rules will be applied synchronously before features are stored ( mode : sync )")
+    .option("--add", "add new entries ( applicable for autotag option only )")
+    .option("--delete", "delete entries ( applicable for autotag option only )")
+    .option("--view", "view entries ( applicable for autotag option only )")
+    // .option("--async", "tag rule will be applied asynchronously after features are written to the storage")
+    // .option("--sync", " [DEFAULT] tag rule will be applied synchronously before features are written to the storage")
+    .action(function (id, options) {
+        tagRuleConfig(id, options).catch((error) => handleError(error))
+    })
+
+async function tagRuleConfig(id: string, options: any) {
+    await common.verifyProBetaLicense();
+    let patchRequest: any = {};
+    let spacedef: any = {};
+    const url = `/hub/spaces/${id}?clientId=cli`
+    const { response, body } = await execute(
+        url,
+        "GET",
+        "application/json",
+        ""
+    );
+    spacedef = body;
+
+    if (spacedef != null) {
+        let ruleTagger = spacedef.processors ? spacedef.processors.find((obj: any) => { return obj.id == 'rule-tagger' }) : null;
+        let ruleTaggerAsync = spacedef.processors ? spacedef.processors.find((obj: any) => { return obj.id == 'rule-tagger-async' }) : null;
+        let taggingRules: any;
+        let taggingRulesAsync: any;
+        if (ruleTagger) {
+            taggingRules = ruleTagger['params'].taggingRules;
+        }
+        if(ruleTaggerAsync) {
+            taggingRulesAsync = ruleTaggerAsync['params'].taggingRules;
+        }
+
+        if (options.delete) {
+            if((!taggingRules || Object.keys(taggingRules).length == 0) && (!taggingRulesAsync || Object.keys(taggingRulesAsync).length == 0)) {
+                console.log("tagrules are not defined for this space yet..!")
+            }
+            else {
+                if (taggingRules && Object.keys(taggingRules).length > 0) {
+                    Object.keys(taggingRules).forEach(key => {
+                        choiceList.push({ 'name': key + " , rule : " + parseJsonPath(taggingRules[key]) + ' , mode : sync', 'value': "sync_" + key });
+                    })
                 }
-           }
+                if (taggingRulesAsync && Object.keys(taggingRulesAsync).length > 0) {
+                    Object.keys(taggingRulesAsync).forEach(key => {
+                        choiceList.push({ 'name': key + " , rule : " + parseJsonPath(taggingRulesAsync[key]) + ' , mode : async  ', 'value': "async_" + key });
+                    })
+                }
```

Comment:
this should be inside else block

Reply:
hi, the idea is to show both sync and async rules in the same list for delete as an agreed feature, so I would prefer and recommend putting this not in else and keep it as is for better user experience. this will let user see all rules in one go.